### ISSUE TITLE
Restructure so that implementing undo and redo is possible too (Can be spun-off into a separate issue too)

### ISSUE THREAD

**Issue by abbyhu2000** [opensearch-project/OpenSearch-Dashboards#3102]

**Comment by joshuarrrr** [comment]
Is this just to unblock redo/undo, or to actually implement those? I thought I remembered that @ashwin-pc had an issue for undo/redo in Vis Builder, but I can't find it, so I must have made that up.

**Comment by ashwin-pc** [comment]
Nope, I didnt actually create that issue, although i agree, this issue needs more context. I dont think undo redo needs a separate restructure task by itself. the whole feature is simple enough that it can be implemented at once once we know the UX for it.
